code/modules/unit_tests/explosion_action.dm

```diff
@@ -0,0 +1,214 @@
+/// Tests the EX_ACT macro on several different types of atoms to ensure that it still works as expected.
+/datum/unit_test/explosion_action
+	/// rolling var for how much brute damage the alien has taken.
+	var/alien_brute_loss = 0
+	/// rolling var for how much burn damage the alien has taken.
+	var/alien_burn_loss = 0
+	/// aliens get a bit of damage done to their ears when exploded, so check that too.
+	var/alien_ear_damage = 0
+
+/datum/unit_test/explosion_action/Run()
+	// We split up this `Run()` into multiple parts based on the over-arching parent type. This is because all of them have different core implementations of `EX_ACT()`, and we want to test all.
+	// All procs also have varying levels of bulkiness to them, and it's valuable to have this level of organization because otherwise it would blend all-together and be an entangled mess.
+	execute_mob_tests()
+	execute_turf_tests()
+	execute_obj_tests()
```

Comment:
should i split this up into different unit tests in the same file? was unsure

Reply:
Meh